src/lib/storage/boot_requirements_strategies/legacy.rb

```diff
@@ -21,18 +21,73 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
-require "yast"
 require "storage/boot_requirements_strategies/base"
-require "storage/planned_volume"
-require "storage/planned_volumes_list"
-require "storage/disk_size"
 
 module Yast
   module Storage
     module BootRequirementsStrategies
       # Strategy to calculate the boot requirements in a legacy system (x86
       # without EFI)
       class Legacy < Base
+
+        using Refinements::Disk
+
+        def needed_partitions
+          volumes = super
+          volumes << grub_volume if grub_partition_required? && grub_partition_missing?
+          if mbr_gap_required?
+            mbr_gap = disk_analyzer.mbr_gap[settings.root_device]
+            # fail if gap is too small
+            if mbr_gap < DiskSize.kiB(256)
+              raise Proposal::Error
```

Comment:
Same than before. `BootRequirementsChecker` does not live inside the `Proposal` namespace. It should not raise `Proposal::Error`

I'm not saying that we should not move `BootRequirementsChecker` into `Proposal` if we decide so.